Packs/CommonPlaybooks/Playbooks/playbook-Search_And_Delete_Emails_-_Generic.yml

```
@@ -126,23 +261,74 @@ inputs:
       accessor: emailfrom
   required: false
   description: The value of the malicious email's "From" attribute.
-  playbookInputQuery:
+  playbookInputQuery: null
 - key: Subject
   value:
     complex:
       root: incident
       accessor: emailsubject
   required: false
   description: The value of the malicious email's "Subject" attribute.
-  playbookInputQuery:
+  playbookInputQuery: null
 - key: AttachmentName
   value:
     complex:
       root: incident
       accessor: attachmentname
   required: false
   description: The value of the malicious email's "AttachmentName" attribute.
-  playbookInputQuery:
+  playbookInputQuery: null
+- key: SearchAndDeleteIntegration
+  value:
+    simple: EWS
+  required: true
+  description: The integration in which to run the search and delete action. Can be
+    O365, or EWS.
+  playbookInputQuery: null
+- key: O365ExchangeLocation
+  value:
+    complex:
+      root: incident
+      accessor: emailto
+  required: false
+  description: Used only in O365. Comma-separated list of mailboxes/distribution groups
+    to include, or use the value "All" to include all.
+  playbookInputQuery: null
+- key: O365KQL
+  value: {}
+  required: false
+  description: Used only in O365. Text search string or a query that is formatted
+    using the Keyword Query Language (KQL).
+  playbookInputQuery: null
+- key: O365Description
+  value: {}
+  required: false
+  description: Used only in O365. Description of the compliance search.
+  playbookInputQuery: null
+- key: |
+    O365AllowNotFoundExchangeLocations
+  value:
+    simple: "false"
+  required: false
+  description: Used only in O365. Whether to include mailboxes other than regular
+    user mailboxes in the compliance search. Default is "false".
+  playbookInputQuery: null
+- key: O365DeleteType
+  value:
+    simple: inputs.O365DeleteType
+  required: false
+  description: Used only in O365. The delete type to perform on the search results.
+    Possible values are Hard and Soft or leave empty to select manually. (Hard = Unrecoverable,
+    Soft=Recoverable)
+  playbookInputQuery: null
+- key: O365ExchangeLocationExclusion
+  value:
+    complex:
+      root: inputs.O365ExchangeLocationExclusion
+  required: false
+  description: Used only in O365. Comma-separated list of mailboxes/distribution groups
+    to exclude when you use the value "All" for the exchange_location parameter.
+  playbookInputQuery: null
 outputs: []
 fromversion: 5.0.0
```

The security and compliance playbook is only available from version 5.5.0
```suggestion
fromversion: 5.5.0
```

---

An important point, I'm on it